### ISSUE TITLE
ZOTS uses DMOVE and DSUB

### ISSUE THREAD

**Issue by larsbrinkhoff** [PDP-10/its#1670]
@eswenson1's recent explorations got me thinking.  Which other program may be using non-KA10 instructions?  Well, ZOTZ is one of them.

**Comment by eswenson1** [comment]
Yeah.  I was going to enter a ticket. I've seen a bunch that are using ADJBP.  I suspect there are other uses of non-KA10 instructions as well.

**Comment by atsampson-abertay** [comment]
Would it be worth patching MIDAS to conditionally emit warnings for non-KA instructions? You could enable this only in the KA10 build, then check for the warnings in the build logs.

**Comment by atsampson** [comment]
Would it be worth patching MIDAS to conditionally emit warnings for non-KA instructions? You could enable this only in the KA10 build, then check for the warnings in the build logs.

**Comment by larsbrinkhoff** [comment]
Excellent idea!  It's like how you can use .MLLIT to get warnings about unterminated literals.

**Comment by larsbrinkhoff** [comment]
The over engineered version would be something like .PROCESSOR==0 means anything goes, 1 for PDP-6, 2 for KA10, 3 for KL10, 4 for KL10 model B, and 5 for KS10.  Or to really go over board, a bit mask.

But really, just warn about non-KA10 instructions.

How about the under engineered version:

EXPUNGE ADJBP,DMOVE,DADD,DSUB,DMUL,... etc

**Comment by eswenson1** [comment]
Not sure I like the under-engineered version. But I do think it should use the logic from KSC; NUUOS to detect the CPU automatically and warn about instructions that aren’t implemented for the detected processor. And perhaps have a way of turning the earning off for those instructions you know there is a UUO handler for (requires a switch to set).

**Comment by larsbrinkhoff** [comment]
That seems like a tall order.  I'd suggest starting with something simple.

I tried patching the DMOVE and DSUB in TIMER, but it complains about negative times.
